### PR TITLE
Cloudfront Integration

### PR DESCRIPTION
**Pull request by njuettner**
Issue: https://github.com/giantswarm/roadmap/issues/1255

## Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/aws/services/iam/oidc.go [hunk 1]
```diff
@@ -13,11 +13,12 @@ import (
 	"github.com/giantswarm/irsa-operator/pkg/util"
 )
 
-func (s *Service) CreateOIDCProvider(bucketName, region string) error {
+func (s *Service) CreateOIDCProvider(domain, bucketName, region string) error {
 	s.scope.Info("Creating OIDC provider")
 
 	s3Endpoint := fmt.Sprintf("s3.%s.%s", region, key.AWSEndpoint(region))
-	identityProviderURL := fmt.Sprintf("https://%s/%s", s3Endpoint, bucketName)
+	//identityProviderURL := fmt.Sprintf("https://%s/%s", s3Endpoint, bucketName)
```
Comment (whites11) [inline pkg/aws/services/iam/oidc.go:20]: Can we remove the comment?

Comment (njuettner) [reply pkg/aws/services/iam/oidc.go:20]: Yeah it was a leftover 🤦🏻

File: pkg/irsa/irsa.go [hunk 2]
```diff
@@ -195,19 +262,19 @@ func (s *IRSAService) Delete(ctx context.Context) error {
...
 		ctrlmetrics.Errors.WithLabelValues(s.Scope.Installation(), s.Scope.AccountID(), s.Scope.ClusterName(), s.Scope.ClusterNamespace()).Inc()
 		s.Scope.Logger.Error(err, "failed to delete S3 bucket")
-		return microerror.Mask(err)
+		return err
 	}
 	err = s.IAM.DeleteOIDCProvider(s.Scope.AccountID(), s.Scope.BucketName(), s.Scope.Region())
 	if err != nil {
 		ctrlmetrics.Errors.WithLabelValues(s.Scope.Installation(), s.Scope.AccountID(), s.Scope.ClusterName(), s.Scope.ClusterNamespace()).Inc()
 		s.Scope.Logger.Error(err, "failed to delete OIDC provider")
-		return microerror.Mask(err)
```
Comment (whites11) [inline pkg/irsa/irsa.go:210]: Why don't we mask any more?

File: pkg/oidc/discovery.go [hunk 3]
```diff
@@ -18,11 +16,13 @@ type DiscoveryResponse struct {
 	ClaimsSupported                  []string `json:"claims_supported"`
 }
 
-func GenerateDiscoveryFile(bucketName, region string) (*bytes.Reader, error) {
+func GenerateDiscoveryFile(domain, bucketName, region string) (*bytes.Reader, error) {
 	// see https://github.com/aws/amazon-eks-pod-identity-webhook/blob/master/SELF_HOSTED_SETUP.md#create-the-oidc-discovery-and-keys-documents
 	v := DiscoveryResponse{
-		Issuer:                           fmt.Sprintf("https://s3.%s.%s/%s", region, key.AWSEndpoint(region), bucketName),
-		JwksURI:                          fmt.Sprintf("https://s3.%s.%s/%s/keys.json", region, key.AWSEndpoint(region), bucketName),
+		//Issuer:                           fmt.Sprintf("https://s3.%s.%s/%s", region, key.AWSEndpoint(region), bucketName),
```
Comment (whites11) [inline pkg/oidc/discovery.go:22]: Please delete leftover comments, thx

File: pkg/irsa/irsa.go (hunk 2 above)
Comment (njuettner) [reply pkg/irsa/irsa.go:210]: The error will be masked in controller.go

File: pkg/oidc/discovery.go (hunk 3 above)
Comment (njuettner) [reply pkg/oidc/discovery.go:22]: 👍🏻

**Review by whites11 — APPROVED** [review]

**Pull request merged**